pandas/core/tools/datetimes.py

```diff
@@ -278,35 +279,41 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
                 yearfirst=yearfirst,
                 require_iso8601=require_iso8601
             )
-            if tz_parsed is not None:
-                if box:
-                    # We can take a shortcut since the datetime64 numpy array
-                    # is in UTC
-                    return DatetimeIndex._simple_new(result, name=name,
-                                                     tz=tz_parsed)
-                else:
-                    # Convert the datetime64 numpy array to an numpy array
-                    # of datetime objects
-                    result = [Timestamp(ts, tz=tz_parsed).to_pydatetime()
-                              for ts in result]
-                    return np.array(result, dtype=object)
+        except ValueError as e:
+            # reached when array_to_datetime raises "unless utc=True..."
+            try:
+                values, tz = conversion.datetime_to_datetime64(arg)
+                return DatetimeIndex._simple_new(values, name=name, tz=tz)
```

Comment:
There _may_ be a future bug here in which the user passes `box=False` and we hit this part and return a DatetimeIndex instead of an array.

I don't exactly know the criteria to set this up, but is it possible to delegate this to `_maybe_box_date_results`?

Reply:
Is #23045 related to the potential bug you have in mind?

> but is it possible to delegate this to _maybe_box_date_results?

It's a little ungainly, but I think it can be done.

Reply:
Not exactly, here's a quick example I found (albeit questionable if this is the correct behavior)

```
In [3]: pd.__version__
Out[3]: '0.24.0.dev0+1010.ge413c491e.dirty'

In [4]: import pytz; import datetime

In [5]: pd.to_datetime([datetime.datetime(2018, 1, 1), datetime.datetime(2018, 1, 1, tzinfo=pytz.UTC)], box=False)
Out[5]: DatetimeIndex(['2018-01-01 00:00:00+00:00', '2018-01-01 00:00:00+00:00'], dtype='datetime64[ns, UTC]', freq=None)
```